packages/beacon-node/src/chain/opPools/attestationPool.ts

```diff
@@ -121,8 +126,14 @@ export class AttestationPool {
       throw new OpPoolError({code: OpPoolErrorCode.REACHED_MAX_PER_SLOT});
     }
 
-    // this should not happen because attestation should be validated before reaching this
-    assert.notNull(committeeIndex, "Committee index should not be null in attestation pool");
+    if (isForkPostElectra(fork)) {
+      // Electra only: this should not happen because attestation should be validated before reaching this
+      assert.notNull(committeeIndex, "Committee index should not be null in attestation pool post-electra");
+      assert.true(isElectraAttestation(attestation), "Attestation should be type electra.Attestation");
```

Comment:
these error are much more concise, should align the ones above in `packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts` accordingly

Reply:
> these error are much more concise, should align the ones above in `packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts` accordingly

The downside of using assert is that statement like `assert.true(isElectraAttestation(attestation), ...)` does not type guard attestation.

https://github.com/ChainSafe/lodestar/blob/4c81a0dec521ba0df1c2960431f984a1573896c4/packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts#L148 in `aggregatedAttestationPool.ts` needs type guard to access `committeeBits` so it's slightly different from `attestationPool`

Reply:
I was mostly talking about the error messages, using assert vs. throwing error depends on the situation. Although we just make sure type inference works with assert the same way